### PR TITLE
Add UUID to ITObject

### PR DESCRIPTION
**Pull request by hamelin**

### CODE REVIEW

File: itsim/__init__.py [hunk 1]
```diff
@@ -11,12 +12,26 @@ class Tag(Tags):
 
 
 class ITObject(TaggedObject):
+
+    def __init__(self, *tags) -> None:
```
Comment (mattcraddock) [inline itsim/__init__.py]: We should add the type annotation here

https://github.com/ElementAI/greensim/blob/39de6a07de998f25a7a398b2ab9dc721db809918/greensim/tags.py#L25

File: tests/test_itsim.py [hunk 2]
```diff
@@ -148,3 +150,14 @@ def test_location_repr():
 def test_location_hash():
     loc = Location("google.ca", 25)
     assert hash(loc) == hash(str(loc))
+
```
Comment (mattcraddock) [inline tests/test_itsim.py]: We should add a test that the `Tags` are instantiated correctly as well

File: itsim/__init__.py (hunk 1 above)
Comment (hamelin) [reply itsim/__init__.py]: asdfasdf

Comment (mattcraddock) [reply itsim/__init__.py]:
```suggestion
    def __init__(self, *tags) -> None:
```

File: tests/test_itsim.py (hunk 2 above)
Comment (hamelin) [reply tests/test_itsim.py]: Out of scope of a unit test. However, I will add a card on the board to revisit tagging of ITObjects, making sure that we get this right for when we need them.

Comment (hamelin) [reply tests/test_itsim.py]: [The card](https://trello.com/c/sSpX6OUk)

**Pull request merged**